tools/code-generation/smithy/codegen/cpp-smoke-tests-codegen/src/main/java/com/amazonaws/util/awsclientsmithygenerator/generators/GenericCodegenAdapter.java

```diff
@@ -0,0 +1,213 @@
+package com.amazonaws.util.awsclientsmithygenerator.generators;
+
+import java.util.List;
+import java.util.Map;
+
+public interface GenericCodegenAdapter<SHAPE, DATA> {
+
+    public Map<String, SHAPE> getMemberShapes(SHAPE shape);
+
+    public SHAPE getShapeFromOperation(String OperationName);
+
+    //public boolean isEnum(DATA d);
+
+    public boolean isFloat(DATA d);
+
+    public boolean isBoolean(DATA d);
+
+    public boolean isInteger(DATA d);
+
+    public boolean isString(DATA d);
+
+    public boolean isMap(DATA d);
+
+    public boolean isList(DATA d);
+
+    public boolean isFloatShape( SHAPE s);
+
+    public boolean isBooleanShape(SHAPE s);
+
+    public boolean isIntegerShape( SHAPE s);
+
+    public boolean isStringShape(SHAPE s);
+
+    public boolean isMapShape(SHAPE s);
+
+    public boolean isListShape( SHAPE s);
+
+    public boolean isEnumShape( SHAPE s);
+
+    public List<DATA> getList(DATA d);
+
+    public Map<String, DATA> getMap(DATA d);
+
+    public SHAPE getListMemberShape(SHAPE list);
+
+    public String getString(DATA d);
+
+    public Boolean getBoolean(DATA d);
+
+    public Float getFloat(DATA d);
+
+    public Integer getInteger(DATA d);
+
+    public String getShapeName(SHAPE s);
+
+    public static String convertSnakeToPascal(String snakeCase) {
```

Comment:
static functions in a interface seems kind of sketchy. Why do they need to be in a interface? they are not being overridden so why can they not live outside of the interface? also why does this function need to exist when untility functions like it exist in [guava](https://github.com/google/guava/wiki/StringsExplained#caseformat) or [apache commons](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html). we shoudlnt be re-implementing common utilities from standard libraries

Reply:
ack. will replace